connector/dynamicroutingconnector/config.go

```diff
@@ -0,0 +1,54 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package dynamicroutingconnector // import "github.com/elastic/opentelemetry-collector-components/connector/dynamicroutingconnector"
+
+import (
+	"errors"
+	"sort"
+	"time"
+
+	"go.opentelemetry.io/collector/pipeline"
+)
+
+type Config struct {
+	DefaultPipelines   []pipeline.ID   `mapstructure:"default_pipelines"`
+	EvaluationInterval time.Duration   `mapstructure:"evalaution_interval"`
+	Pipelines          [][]pipeline.ID `mapstructure:"pipelines"`
+	Thresholds         []int           `mapstructure:"thresholds"`
```

Comment:
Threshold directly linking to pipeline id also seems like a vialble option. ["10": "traces/low"]